pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -441,3 +448,37 @@ func (OutboundProxyGenerator) determineRoutes(
 
 	return routes
 }
+
+func buildServiceToVipsMap(outbounds []*mesh_proto.Dataplane_Networking_Outbound, meshVIPOutbounds []*mesh_proto.Dataplane_Networking_Outbound) (map[string]bool, map[string][]mesh_proto.OutboundInterface) {
+	vipMap := map[string][]mesh_proto.OutboundInterface{}
+	var dpNetworking *mesh_proto.Dataplane_Networking
+
+	serviceVips := map[string]bool{}
+	for _, ob := range meshVIPOutbounds {
+		service := ob.GetService()
+		if service != "" {
```

Comment:
IMO checks like this are just noise over the algorithm, we can assume a non empty service here.

Reply:
I've seen this in somewhere else, so I just added.